addons/resource/views/resource_views.xml

```diff
@@ -159,9 +160,52 @@
         <field name="context">{
             'default_calendar_id': active_id,
             'search_default_calendar_id': active_id}</field>
+        <field name="domain">[('time_type', 'in', ['leave', 'other'])]</field>
         <field name="search_view_id" ref="view_resource_calendar_leaves_search"/>
     </record>
 
+    <record id="action_open_resource_calendar_leave_tree" model="ir.actions.act_window.view">
```

Comment:
`resource_calendar_leaves_action_from_calendar_view_<view_type>` might be a better format. We should add this point in the guidelines ;) @tde-banana-odoo What do you think, as your advice on such spiritual question is essential.